### ISSUE TITLE
No humidity control

### ISSUE THREAD

**Issue by highincolorado** [PeteRager/lennoxs30#149]
Just had a steam humidifier installed and find that the humidity setpoint in HA is not agreeing with what the system setpoint is and an error is thrown when trying to set new setpoint via HA. 

“Failed to call service climate/set_humidity” flashes on screen when a humidity setpoint change is attempted.

From the log:
File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 454, in async_set_humidity
await self.hass.async_add_executor_job(self.set_humidity, humidity)
File "/usr/local/lib/python3.9/concurrent/futures/thread.py", line 58, in run
result = self.fn(*self.args, **self.kwargs)
File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 450, in set_humidity
raise NotImplementedError()
NotImplementedError

**Comment by PeteRager** [comment]
Yes, that part of the integration is not tested / implemented.  That said we should be able to get it working.  I have one high priority bug fix I need to get out, then I’ll start looking at this for the next release.   We’ll likely need to capture message logs.  

Once you are setup with message logging, go turn the humidifier on and off and adjust its setpoint from the S30 panel and let me know what values you set it to.

Easiest way to get message logging setup up is in the integration configuration, select the log message to file option and type in a file name “s30_message.log”, reload integration or restart.  Then post or email the file (it’ll be in the same folder as the home assistant log). It’d be good to have the full message log so I can see what’s in the initial equipment configuration, as that may have range limits and information of the model of steam humidifier for your device configuration.

**Comment by highincolorado** [comment]
I’ll get on that tomorrow I hope, busy day today.
________________________________

**Comment by highincolorado** [comment]
Ok I got logging set up. Turned the humidity off then back on. Setpoint was at 35 changed it to 40 then back to 35.

I made the changes from my office using the lenox app on my iPad not directly from the thermostat, if this makes a difference let me know and I’ll redo the capture.

Rick

**Comment by PeteRager** [comment]
Doing it from the iPad app is fine.

If you replied to the GitHub email with the log file as an attachment that does not go through.  

Post it directly from the GitHub website or send to my email - 

pete dot sage
at icloud dot com

**Comment by PeteRager** [comment]
I do not see the message log yet, please post when you have time.

**Comment by highincolorado** [comment]
Sorry, fat fingers at play. here it is. Let me know what else I can do to help.

[s30_message.log](https://github.com/PeteRager/lennoxs30/files/8592503/s30_message.log)

**Comment by PeteRager** [comment]
Perfect, has all the info we need.  I can see the different operations.  I'll get working on it.

**Comment by PeteRager** [comment]
I’m making progress on this.  The code is mostly written and next step is to test on simulator.   My system does not have humidification, it just has dehumidification.  The Lennox implementation looks the similar so I’ve added both and then will test dehumdification on/off and humidity setpoint on my system.  Once that is done, I’ll make a release for you to test the humidification.   I think we are a week out.  Aiming for 5/13. 

A question, the dehumidification settings on my unit  have a couple of modes - there is a “Normal” and “Max” and when Max is selected there is an overcooling setpoint which ranges from 1-4 degrees F - which I’ll add support for.  Is there a similar set of settings on your unit for humidification?

**Comment by highincolorado** [comment]
Here are screen shots of the available humidity settings:

![E63C2AC2-9BC8-46BE-AB50-4EB7DC76033C](https://user-images.githubusercontent.com/60855018/167127033-8b024899-ea0b-48b7-8320-6b6d15df30ff.jpeg)
![36D4C396-7418-4F70-A7B0-52A0F5A0D6C3](https://user-images.githubusercontent.com/60855018/167127066-af707814-7b4d-4edd-9a94-8fa9b5991646.jpeg)

**Comment by PeteRager** [comment]
Thanks

**Comment by PeteRager** [comment]
I've tested the changes with a simulator and built this lovelace dashboard.  The climate entity now has an attribute called **humidityMode** - that is what is being displayed in the second row.  This indicates if dehumidification or humidification is turned of for the zone.  There is a separate attribute called **humidityOperation** (not shown) that indicates if the humidity mode is current running (for example humidifying).  In the third row, cards are displaying the **Humidity** attribute of the climate entity (also called Target Humidity).  Lennox has separate humidify and dehumidify setpoints, so this attributes shows the value based on the current humidity mode.  If humidity mode is Off there is no target.

The fourth set of entities all in one card are new **select** entities that are used to set the humidityMode for the zone.  The options available depend on the equipment installed and they reflect the current humidityMode.

![humidify](https://user-images.githubusercontent.com/76050312/168175788-187ec85a-e7a6-41a2-8dd8-59d3e977152d.PNG)

Here's a look at the dropdown on one of the select entities, because the simulator has both dehumidification and humidication capability those options are available, as well as an off value.

![humoperation](https://user-images.githubusercontent.com/76050312/168175794-d6187603-385d-4f94-a124-0f8a77d3bb34.PNG)

This should meet your needs?  Feedback welcome.

Am running couple days behind.

**Comment by highincolorado** [comment]
Looks like more than this beggar had a right to expect.

Thanks

**Comment by PeteRager** [comment]
Code is done and tested.  When I close the pull request to create the release it will close this issue.  New release 0.2.6 should be available by tomorrow.   I have tested the dehumidify side of stuff as that’s what I have.  I’m pretty confident the Humidify stuff should work.  If it doesn’t create a new issue.

**Comment by PeteRager** [comment]
released
https://github.com/PeteRager/lennoxs30/releases/tag/v0.2.6

**Comment by highincolorado** [comment]
Installed, I now see the humidity setpoint and can adjust it. No errors that I see in the logs.

Thanks

**Comment by PeteRager** [comment]
👍
